tests/test_halo_update.py

```diff
@@ -305,12 +305,12 @@ def test_halo_update_timer(
     """
     test that halo update produces nonzero timings for all expected labels
     """
-    req_list = []
+    halo_updater_list = []
```

Comment:
Can you add a test somewhere to this file (or more likely to a new file, since you don't need the very complicated fixtures in this one) that when you use the halo updater multiple times in sequence, it behaves as expected? Particularly that the halos are updated identically each time (correctness is already tested by this file so you don't need to re-test it) and that buffers are/aren't re-allocated in the way you expect based on the arguments?

Reply:
I added a test in test_halo_update. To properly test I needed to do a full halo update, albeit a small one, and that meant copying around a bunch of fixtures. Felt like it was less clunky to specialize a test in this file rather than create a new test file. Alternatively I could try and share more of the testing fixture across the test base

Reply:
Doing the test here is OK. I just thought you don't need to worry about the input/output data as much because you just need the outputs to be the same and not necessarily to make sense, but yeah most of the fixtures are to set up the Communicator.